static/app/views/alerts/rules/metric/details/body.tsx

```diff
@@ -177,7 +182,13 @@ export default function MetricDetailsBody({
             showAbsolute={false}
             disallowArbitraryRelativeRanges
           />
-
+          {onDemandMetricAlert && isEmptyAlert && (
+            <OnDemandMetricAlert
+              message={t(
+                'This alert lacks historical data due to filters for which we don’t routinely extract metrics.'
+              )}
+            />
+          )}
```

Comment:
No super hard opinion on this, but I wonder if this should rather be a functionality of the `MetricChart` so we don't have to break out of the uni-directional data flow. 🤔 
It could support an `emptySeriesWarning: ReactNode` prop.

Reply:
yep, this made it much cleaner! Thanks!